OpenRA.Mods.Common/Traits/BotModules/Squads/States/GroundStates.cs

```diff
@@ -25,6 +25,25 @@ protected Actor FindClosestEnemy(Squad owner)
 		{
 			return owner.SquadManager.FindClosestEnemy(owner.Units.First().CenterPosition);
 		}
+
+		protected Actor GetRandomValuableTarget(Squad owner)
+		{
+			var manager = owner.SquadManager;
+			var mustDestroyedEnemy = manager.World.ActorsHavingTrait<MustBeDestroyed>(t => t.Info.RequiredForShortGame)
+					.Where(a => manager.IsPreferredEnemyUnit(a) && manager.IsNotHiddenUnit(a)).ToArray();
+
+			if (!mustDestroyedEnemy.Any())
+				return FindClosestEnemy(owner);
+
+			return mustDestroyedEnemy.Random(owner.World.LocalRandom);
+		}
+
+		protected Actor ThreatScan(Squad owner, Actor teamLeader, WDist scanRadius)
```

Comment:
This function is mainly focus on return a relatively reasonable target since this is a regular groun attack squad on F2A. Short game target is not equal as the closest target, and heading to Short game target can always pose the directly threat to player. 

Remember GroundAttackMoveState is heading to target mainly, heading to a structure that deep inside enemy base will lead squad find the base, and convert to attack state when they have first encounter.